devicetypes/smartthings/zwave-radiator-thermostat.src/zwave-radiator-thermostat.groovy

```diff
@@ -121,7 +123,24 @@ def zwaveEvent(physicalgraph.zwave.commands.securityv1.SecurityMessageEncapsulat
 	}
 }
 
+def zwaveEvent(physicalgraph.zwave.commands.multicmdv1.MultiCmdEncap cmd) {
+	cmd.encapsulatedCommands().collect { encapsulatedCommand ->
+		zwaveEvent(encapsulatedCommand)
+	}.flatten()
+}
 
+def zwaveEvent(physicalgraph.zwave.commands.wakeupv2.WakeUpNotification cmd) {
+	def cmds = []
+	if (!isPoppRadiatorThermostat()) {
+		cmds += zwave.batteryV1.batteryGet() // POPP sends battery report automatically every wake up by itself, there's no need to duplicate it
+	}
+	cmds += [
+			zwave.thermostatSetpointV2.thermostatSetpointSet([precision: 1, scale: 0, scaledValue: state.cachedSetpoint, setpointType: 1, size: 2]),
+			zwave.thermostatSetpointV2.thermostatSetpointGet(setpointType: 1),
+			zwave.wakeUpV2.wakeUpNoMoreInformation()
```

Comment:
@kianooshST here's some command caching for wakeup like we had talked about. Maybe a more generic implementation in the future?